src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // Set the amount avaiable line edit to the amount available amount in the config file, rounded to two decimal places
+        m_amountAvailableEdit->setText(QString::number(m_totalAmountAvailable, 'f', 2));
+
+        // Create the table widget using the bill map's contents
+        createTableWidgetUsingMap();
+
+        // Display the main window
+        this->show();
+    }
+}
+
+void MainWindow::attemptConfigFileGeneration()
+{
+    // Create a directory object
+    QDir configParentFolder;
+
+    // If the config file path could not be generated
+    if(!configParentFolder.mkpath(m_CONFIG_PARENT_FOLDER))
+    {
+        // Display a message box to alert the user
+        createConfigFileGenerateFailureBox();
+    }
+
+    // The config file was found in the expected path
+    else if(QFile::exists(m_CONFIG_FILE_DIRECTORY))
+    {
+        // Read config file and create UI
+        readConfigAndCreateUI();
+
+    }
+
+    // The config file path exists but the file is not in it
+    else
+    {
+        // Prompt user for financial information
+        welcomeFirstTimeUser();
+    }
+}
+
+
+void MainWindow::welcomeFirstTimeUser()
+{
+    // Create a welcome message box with appropriate title and title
+    QMessageBox welcomeBox;
+    welcomeBox.setText(m_WELCOME_BOX_PRIMARY_TEXT);
+    welcomeBox.setInformativeText(m_WELCOME_BOX_INFO_TEXT);
+
+    // Allow options for advancing to the next screen or closing the application
+    welcomeBox.setStandardButtons(QMessageBox::Ok | QMessageBox::Close);
+
+    // Save the user's selection
+    int welcomeBoxSelection = welcomeBox.exec();
+
+    switch(welcomeBoxSelection)
+    {
+        // If the user selects "Ok"
+        case QMessageBox::Ok :
+        {
+            // Ask for total funds available
+            askForTotalAmountAvailable();
+            break;
+        }
+
+        // If the user selects "Close"
+        case QMessageBox::Close :
+        {
+            // Exit application
+            emit conditionToTerminateMet();
+            break;
+        }
+
+        // Break in the default case
+        default:
+        {
+            break;
+        }
+    }
+}
+
+void MainWindow::createCorruptConfigFileBox()
+{
+    // Create the corrupt config file QMessageBox with appropriate displayed text
+    QMessageBox corruptBox;
+    corruptBox.setText(m_CONFIG_CORRUPT_FILE_BOX_PRIMARY_TEXT);
+    corruptBox.setInformativeText(m_CONFIG_CORRUPT_FILE_BOX_INFO_TEXT);
+
+    // Give the user the options of "Ok'ing" the error or closing the application
+    corruptBox.setStandardButtons(QMessageBox::Ok | QMessageBox::Close);
+
+    // Store the user's selection
+    int corruptBoxSelection = corruptBox.exec();
+
+    switch(corruptBoxSelection)
+    {
+        // If the user selected Ok
```

Comment:
In the previous function, the button selections were wrapped in quotation marks. Do one or the other.

Reply:
Comments made consistent.